src/main/scala/uk/gov/nationalarchives/aggregate/processing/modules/TransferOrchestration.scala

```diff
@@ -0,0 +1,65 @@
+package uk.gov.nationalarchives.aggregate.processing.modules
+
+import cats.effect.IO
+import com.typesafe.scalalogging.Logger
+import graphql.codegen.types.ConsignmentStatusInput
+import uk.gov.nationalarchives.aggregate.processing.config.ApplicationConfig.getClientSecret
+import uk.gov.nationalarchives.aggregate.processing.graphql.GraphQlApi
+import uk.gov.nationalarchives.aggregate.processing.graphql.GraphQlApi.{backend, keycloakDeployment}
+import uk.gov.nationalarchives.aggregate.processing.modules.Common.ConsignmentStatusType
+import uk.gov.nationalarchives.aggregate.processing.modules.Common.ProcessType.AssetProcessing
+import uk.gov.nationalarchives.aggregate.processing.modules.Common.StateStatusValue.{Completed, CompletedWithIssues, ConsignmentStatusValue, Failed}
+import uk.gov.nationalarchives.aggregate.processing.modules.ErrorHandling.BaseError
+import uk.gov.nationalarchives.aggregate.processing.modules.TransferOrchestration.{AssetProcessingEvent, OrchestrationResult, TransferError}
+
+import java.util.UUID
+import scala.util.{Failure, Success, Try}
+
+class TransferOrchestration(graphQlApi: GraphQlApi)(implicit logger: Logger) {
+
+  def orchestrate[T <: Product](orchestrationEvent: T): Try[IO[OrchestrationResult]] = {
+    orchestrationEvent match {
+      case assetProcessingEvent: AssetProcessingEvent => Success(orchestrateProcessingEvent(assetProcessingEvent))
+      case _                                          => Failure(throw new RuntimeException(s"Unrecognized orchestration event: ${orchestrationEvent.getClass.getName}"))
+    }
+  }
+
+  private def orchestrateProcessingEvent(event: AssetProcessingEvent): IO[OrchestrationResult] = {
+    val errors = event.processingErrors
+    val consignmentId = event.consignmentId
+    val consignmentStatusValue: ConsignmentStatusValue = if (errors) { Failed }
+    else Completed
+
+    if (errors) {
+      val transferError = TransferError(consignmentId, s"$AssetProcessing.$CompletedWithIssues", "One or more assets failed to process.")
+      ErrorHandling.handleError(transferError, logger)
```

Comment:
Just double checking but we don't update the consignment status with `completedWithIsssues` it's either `completed` or `failed`?

Reply:
This isn't the consignment status. This is the error message. The asset processing did all complete, but it errors were found with the assets that were processed. `Failed` means the processing as a whole didn't work, ie it failed to run.